erpnext/accounts/doctype/gl_entry/gl_entry.py

```diff
@@ -186,7 +189,6 @@ def validate_dimensions_for_pl_and_bs(self):
 				and self.company == dimension.company
 				and dimension.mandatory_for_pl
 				and not dimension.disabled
-				and not self.is_cancelled
```

Comment:
This was intentionally added to handle edge case: Dimension is made mandatory after a vouchers was created. Let it be there.